ui/src/app/tags/components/TagsHeader/TagsHeader.tsx

```diff
@@ -0,0 +1,39 @@
+import { Button } from "@canonical/react-components";
+
+import MachinesHeader from "app/base/components/node/MachinesHeader";
+import TagHeaderForms from "app/pools/components/TagHeaderForms";
+import { TagHeaderViews } from "app/tags/constants";
+import type { TagHeaderContent, TagSetHeaderContent } from "app/tags/types";
+
+type Props = {
+  headerContent: TagHeaderContent | null;
+  setHeaderContent: TagSetHeaderContent;
+};
+
+export const TagsHeader = ({
+  headerContent,
+  setHeaderContent,
+}: Props): JSX.Element => {
+  return (
+    <MachinesHeader
+      buttons={[
+        <Button
+          appearance="positive"
+          onClick={() => setHeaderContent({ view: TagHeaderViews.AddTag })}
+        >
+          Create new tag
+        </Button>,
+      ]}
```

Comment:
This relates to my other comment about view hierarchy.

I think that instead of importing a parent element into a child it would make more sense to keep MachineHeader above and in a single place (`views/Machines`), like it was before.

Reply:
The problem we have is that other than the title and the tabs we treat the machines/tags/pools as isolated parts of the site (there's no shared state etc.) so we're having to work around that by watching for URL changes and resetting things like the filter states.

As we're now adding forms for the tags we're now conflating things further. The aim of this PR is separating the concerns of each view (machines/tags/pools) so that they no longer have a single header and form state so each section can implement things like forms in isolation.